flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -908,38 +909,56 @@ public void reportCheckpointMetrics(
         // will be restarted by the CheckpointCoordinatorDeActivator.
         checkpointCoordinator.stopCheckpointScheduler();
 
+        final CompletableFuture<Collection<ExecutionState>> executionGraphTerminationFuture =
+                FutureUtils.combineAll(
+                        StreamSupport.stream(
+                                        executionGraph.getAllExecutionVertices().spliterator(),
```

Comment:
Just a question, we could theoretically iterate over regions instead of all vertices, here, right?

Reply:
I'm not that familiar with the regions concept, yet. AFAIK, regions are collections of tasks that have to run together, right? Do they share the same `ExecutionState`? I guess, @tillrohrmann could answer that question.

Reply:
I don't think we have access to the pipelined regions in SchedulerBase (they are in the SchedulingStrategy of the DefaultScheduler). Secondly, the Regions don't have termination futures.

Disclaimer: My experience with this part of the flink code base is based on reading the code the last few minutes.

Reply:
Yeah, it took me also some digging to find a connection. I found  `executionGraph.getSchedulingTopology().getAllPipelinedRegions()`. Not sure, though, whether this works conceptually.